src/main/java/entities/default_game/IDrawOutputBoundary.java

```diff
@@ -0,0 +1,42 @@
+package entities.default_game;
+
+import java.awt.*;
+import java.awt.image.BufferedImage;
+
+/**
+ * The output boundary for drawing to the window.
+ * The presenter can implement this interface, and any entities
+ * which need drawing can have a draw method which takes this.
+ */
+public interface IDrawOutputBoundary {
+    /** Get the size of a tile in pixels */
+    int getTileSize();
+
+    /** Draw an image.
+     *
+     * @param image The image to be drawn.
+     * @param x The X position (in pixels) to draw the image at.
+     * @param y The Y position (in pixels) to draw the image at.
+     * @param width The width (in pixels) that the image should be drawn with.
+     *              (If this is different from the image's width, it will be scaled appropriately)
+     * @param height The height (in pixels) that the image should be drawn with.
+     */
+    void drawImage(BufferedImage image, int x, int y, int width, int height);
+
+    /** Draw a rectangle
+     *
+     * @param x The X position (in pixels) of the top-left corner of the rectangle.
+     * @param y The Y position (in pixels) of the top-left corner of the rectangle.
+     * @param width The width (in pixels) of the rectangle.
+     * @param height The height (in pixels) of the rectangle.
+     * @param color The color of the rectangle.
+     */
+    void drawRect(int x, int y, int width, int height, Color color);
+}
+
+// Note: If you have a java.awt.Graphics2D object g, you can implement
```

Comment:
Like Seamus suggested, I think we could take Graphics2D g2 as an input instead so that we don't have to rely on the fact they do/don't have Graphics2D.

Reply:
Sure. I guess Graphics2D is already an abstract class, so we're not depending on any particular graphics implementation.

Reply:
The only problem is that I'm not sure whether I should use TILE_SIZE  from the entities layer. Is it okay to do that because it's a constant?

Reply:
Actually, TILE_SIZE isn't public, so it's impossible to use it from the entities layer. There needs to be some way of passing in the tile size. Do you think there should be an extra tileSize parameter for the draw method?

Reply:
That's also what I was thinking. Many classes in different layers use TILE_SIZE, MAX_PANEL_ROW etc. 
I wonder if there's a way to store these "global constants" that any class can access, without breaking CA rule...

Reply:
I agree with using a getter method to obtain the tilesize, but the problem is that we would need to define it in many classes although tilesize is just a constant... There should be a way to store these constants, perhaps we could use a hash map in the use cases layer?

Reply:
If we implement a hash map for the game, we can use it for the different cases used in custom asset setter as well.